svm/src/transaction_processing_result.rs

```diff
@@ -87,4 +87,17 @@ impl ProcessedTransaction {
             Self::FeesOnly { .. } => None,
         }
     }
+
+    pub fn executed_units(&self) -> u64 {
+        self.execution_details()
+            .map(|detail| detail.executed_units)
+            .unwrap_or_default()
+    }
+
+    pub fn loaded_accounts_data_size(&self) -> u32 {
+        match self {
+            Self::Executed(context) => context.loaded_transaction.loaded_accounts_data_size,
+            Self::FeesOnly(details) => details.rollback_accounts.data_size() as u32,
+        }
+    }
```

Comment:
@anza-xyz/svm: could you review this tiny code addition to your land? (maybe @Lichtso ?)

some context:

in this pr, i made some operation (namely, `check_block_cost_limits()`) to work on `TransactionProcessingResult`, not on `TransactionCommitResult`. That operation only needed these two fields and they can readily be derived from `TransactionProcessingResult` as shown. Hence, these methods are added to `TransactionProcessingResult` and accordingly `TransactioCommitResult` construction code (`create_commit_results()`) is adjusted as well not to duplicate same business logic.

to begin with, that operation's timing change is desired for minor optimization: https://github.com/anza-xyz/agave/pull/4150#discussion_r1903755918

Reply:
`executed_units()` is only an alias to an already public getter method?

`loaded_accounts_data_size()` looks ok, but keep in mind that there is a known bug in `loaded_accounts_data_size` not counting loader-v3 programs properly.

Reply:
@jstarry may be the relevant person here IIRC; the `FeesOnly` and `ProcessingResult` stuff is to support SIMD-0191 (formerly part of SIMD-0082). He was the implementor of that code, and I reviewed.

He should take a look as well, but to me this looks correct.

Reply:
@anza-xyz/svm: could you review this again?

I needed to push another commit to fix a bug, which is unrelated to the changes of the `svm` subtree: https://github.com/anza-xyz/agave/pull/4150#discussion_r1909677298